tests/gordo_components/client/test_client.py

```diff
@@ -443,90 +440,52 @@ def test_ml_server_dataframe_to_dict_and_back(tags: typing.List[str]):
+        config={
+            "name": name,
+            "dataset": {
+                "tags": ["tag-1", "tag-"],
+                "train_start_date": "2016-01-01T00:00:00Z",
+                "train_end_date": "2016-01-05T00:00:00Z",
+            },
+            "model": "sklearn.linear_model.LinearRegression",
+        },
+        project_name="test-project",
+    )
+
+
 @pytest.mark.parametrize(
-    "endpoints,target,ignore_unhealthy,expected",
+    "machines,target,expected",
     [
-        # One unhealthy + one healthy + no target + ignoring unhealthy = OK
-        (
-            [_endpoint_metadata("t1", False), _endpoint_metadata("t2", True)],
-            None,
-            True,
-            [_endpoint_metadata("t2", True)],
-        ),
-        # One unhealthy + one healthy + no target + NOT ignoring unhealthy = ValueError
-        (
-            [_endpoint_metadata("t1", False), _endpoint_metadata("t2", True)],
-            None,
-            False,
-            ValueError,
-        ),
-        # One unhealthy + one healthy + target (healthy) + do not ignore unhealthy = OK
-        # because the client doesn't care about the unhealthy endpoint with a set target
-        (
-            [_endpoint_metadata("t1", False), _endpoint_metadata("t2", True)],
-            "t2",
-            False,
-            [_endpoint_metadata("t2", True)],
-        ),
-        # All unhealthy = ValueError
-        # ...even if we're suppose to ignore unhealthy endpoints, because Noen are healthy
-        (
-            [_endpoint_metadata("t1", False), _endpoint_metadata("t2", False)],
-            None,
-            True,
-            ValueError,
-        ),
-        # All unhealthy + ignore unhealthy = ValueError
-        # ...because we end up with no endpoints
-        (
-            [_endpoint_metadata("t1", False), _endpoint_metadata("t2", False)],
-            None,
-            True,
-            ValueError,
-        ),
-        # All healthy = OK
-        (
-            [_endpoint_metadata("t1", True), _endpoint_metadata("t2", True)],
-            None,
-            True,
-            [_endpoint_metadata("t1", True), _endpoint_metadata("t2", True)],
-        ),
-        # All healthy + target = OK (should get back only that target)
-        (
-            [_endpoint_metadata("t1", True), _endpoint_metadata("t2", True)],
-            "t2",
-            True,
-            [_endpoint_metadata("t2", True)],
-        ),
-        # Want to filter down to one target, but that target is not healthy, ValueError
-        (
-            [_endpoint_metadata("t1", True), _endpoint_metadata("t2", False)],
-            "t2",
-            True,
-            ValueError,
-        ),
+        # Two machines, no target, should give two machines
+        ([_machine("t1"), _machine("t2")], None, [_machine("t1"), _machine("t2")]),
+        # One machine target should filter down to that machine
+        ([_machine("t1"), _machine("t2")], "t2", [_machine("t2")]),
```

Comment:
Sweet. Would be cool to have some advanced querying/regex matching in the future.